[WIP] DOC Explain missing value mechanisms (#23746)
scikit-learn/scikit-learn: doc/modules/impute.rst

:ref:`glossary` entry on imputation.

Missing value mechanisms
========================

> Three mechanisms model data missingness.

Reviewer: ```suggestion
Three mechanisms model data missingness exist:
```

> Three mechanisms model data missingness.

Reviewer 2: ```suggestion
The machine learning literature typically distinguishes between the following
settings. Note that the names are not necessarily very intuitive:
```

> * **Missing Completely At Random (MCAR)**: the missingness does not depend on data.

Reviewer: What about giving a concrete example for each mechanism to illustrate it.

* **Missing At Random (MAR)**: the missingness does not depend on underlying
  missing values but can depend on observed ones.
* **Missing Not At Random (MNAR)**: the missingness depends on underlying missing

>   values.

Reviewer 2: ```suggestion
  values. Therefore, the missingness pattern can be statistically associated
  with `y` in a supervised classification or regression setting.
```

.. figure:: ../images/missing_value_mechanisms.png
   :align: center
   :scale: 20%

> In the above example, X1 is always observed. In the first plot, X2 is masked

Reviewer: ```suggestion
In the above example, X1 is always observed. In the left-hand side plot, X2 is masked
```

independently of the values of (X1, X2), hence MCAR. In the second, X2 is

> masked when X1 (observed) reaches some threshold, hence MAR. In the last, X2 is

Reviewer: ```suggestion
masked when X1 (observed) reaches some threshold, hence MAR. In the right-hand side plot, X2 is
```
